docs/interop-kotlinx-metadata.md

```diff
@@ -35,21 +35,49 @@ println("Is class? ${kmClass.isClass}")
 println("Is data class? ${kmClass.isData}")
 ```
 
-### Interop with kotlinx-metadata
+### Interop with KotlinPoet
+
+`interop:kotlinx-metadata` offers an API for converting core kotlinx-metadata `Km` types to
+KotlinPoet source representations of their APIs. This includes full type resolution, signatures,
+enclosed elements, and general stub source representations of the underlying API.
 
-To convert a kotlinx-metadata type to its analogous immutable type, simply call `toImmutable()`
+### Example
 
 ```kotlin
-val immutableKmClass: ImmutableKmClass = kmClass.toImmutable()
-```
+data class Taco(val seasoning: String, val soft: Boolean) {
+  fun prepare() {
+  }
+}
 
-To convert an immutable type back to its analogous kotlinx-metadata type, call `toMutable()`.
+val typeSpec = Taco::class.toTypeSpec()
 
-```kotlin
-val mutableKmClass: KmClass = kmClass.toMutable()
+// Or FileSpec
+val fileSpec = Taco::class.toFileSpec()
 ```
 
-### Interop with KotlinPoet
+### Source representation
+
+The generated representations are a _best effort_ representation of the underlying source code.
+This means that synthetic elements will be excluded from generation. Kotlin-specific language
+features like lambdas or delegation will be coerced to their idiomatic source form.
+
+To aid with this, `toTypeSpec()` and `toFileSpec()` accept optional `ClassInspector` instances
+to assist in parsing/understanding the underlying JVM code. This is important for things like
+annotations, companion objects, certain JVM modifiers, overrides, and more. While it is optional,
+represented sources can be incomplete without this information available. Reflective and javax
+`Elements` implementations are available under the `kotlinpoet-classinspector-*` artifacts.
```

Comment:
All class inspectors are in this artifact now, need to reword this statement?

Reply:
f4b001b